File: ios/Tehilim/Features/Chains/ChainCountdown.swift
import Foundation

/// Compte à rebours partagé des chaînes. Format `HH:MM:SS` (préfixé des jours si
/// besoin) → les **secondes défilent** chaque tick, contrairement à un affichage
/// « 5 h 23 min » qui semblait figé.
enum ChainCountdown {
    static func format(seconds: Int) -> String {
        let secs = max(0, seconds)
        let d = secs / 86400, h = (secs % 86400) / 3600, m = (secs % 3600) / 60, s = secs % 60
        let hh = String(format: "%02d", h), mm = String(format: "%02d", m), ss = String(format: "%02d", s)
        let dU = AppLocale.code == "en" ? "d" : "j"
        return d > 0 ? "\(d)\(dU) \(hh):\(mm):\(ss)" : "\(hh):\(mm):\(ss)"
    }
}

File: android/app/src/main/java/com/david/tehilim/features/chains/ChainCountdown.kt
package com.david.tehilim.features.chains

/**
 * Compte à rebours partagé des chaînes. Format `HH:MM:SS` (préfixé des jours si
 * besoin) → les secondes défilent à chaque tick, contrairement à un affichage
 * « 5 h 23 min » qui semblait figé.
 */
internal fun chainCountdown(ms: Long): String {
    val total = (ms / 1000).coerceAtLeast(0)
    val d = total / 86400
    val h = (total % 86400) / 3600
    val m = (total % 3600) / 60
    val s = total % 60
    val hms = "%02d:%02d:%02d".format(h, m, s)
    return if (d > 0) "${d}j $hms" else hms
}
